gpu_bdb/bdb_tools/sessionization.py

```diff
@@ -14,15 +14,18 @@
 # limitations under the License.
 #
 
-import cupy as cp
 import numpy as np
 
-
-def get_session_id_from_session_boundry(session_change_df, last_session_len):
+def get_session_id_from_session_boundry(session_change_df, last_session_len, cpu=False):
     """
         This function returns session starts given a session change df
     """
-    import cudf
+    if cpu:
+        import pandas as cudf
```

Comment:
Given the name of this repository, it was clearly written initially to be run on GPUs.

Now that we're looking into expanding the benchmark to support Dask DataFrame on CPUs, a general question is whether we actually want to:

1. import CPU libraries as if they're GPU libraries, ex: `import pandas as cudf`
2. inverse it, and import GPU libraries as if they're CPU libraries, ex: `import cudf as pd`
3. Do something like:
```
if cpu:
  from pandas import DataFrame
else:
  from cudf import DataFrame
```

Interested in opinions here

Reply:
I think #2 conveys the idea that existing CPU code might be accelerated without changing more than an import, which is nice.

#3 feels cleaner, although using the `DataFrame` class directly is probably substantially less common usage than `pd.DataFrame`/`cudf.DataFrame`.

Reply:
How often do we need to do this, and for what kinds of operations? If it's frequent but mostly for a select few types of operations, perhaps we might consider wrapping those operations into "safe" helper functions (or even putting them into a registry that dispatches appropriately)?

We've often done something like that with Dask CPU/GPU compatibility in the past.

https://github.com/dask/dask/blob/a81fed43830c32f836a988a07c10ca8f99279bec/dask/array/linalg.py#L973-L974

https://github.com/dask/dask/blob/a81fed43830c32f836a988a07c10ca8f99279bec/dask/array/utils.py#L568-L585